processor/download_test.go

```diff
@@ -52,6 +52,54 @@ func TestPerformUserAgent(t *testing.T) {
 	wg.Wait()
 }
 
+func TestPerformJobUserAgent(t *testing.T) {
+	var wg sync.WaitGroup
+
+	//Since we are messing with the default settings, we create a new processor here
+	p, err := New(store, 1, storageDir, logger)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	p.UserAgent = "Downloader Processor UserAgent"
+	a, err := job.NewAggregation("FooBar", 1, "")
+	if err != nil {
+		log.Fatal(err)
+	}
+	wp, err := p.newWorkerPool(*a)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	j := job.Job{ID: "TestUserAgent",
+		URL:         strings.Join([]string{server.URL, t.Name()}, "/"),
+		AggrID:      a.ID,
+		CallbackURL: "http://example.com",
```

Comment:
Lets start using the new fields `CallbackType` and `CallbackDst` here!

Reply:
As discussed, lets change this all together when CallbackURL gets discarded.